src/Database/Esqueleto/Internal/Internal.hs

```diff
@@ -401,7 +404,10 @@ having expr = Q $ W.tell mempty { sdHavingClause = Where expr }
 --
 -- @since 2.2.7
 locking :: LockingKind -> SqlQuery ()
-locking kind = Q $ W.tell mempty { sdLockingClause = Monoid.Last (Just kind) }
+locking kind = putLocking $ GeneralLockingClause kind
+
+putLocking :: LockingClause -> SqlQuery ()
+putLocking clause = Q $ W.tell mempty { sdLockingClause = clause }
```

Comment:
Please add a doc comment 

```suggestion
-- |
--
-- @since {this version}
putLocking :: LockingClause -> SqlQuery ()
putLocking clause = Q $ W.tell mempty { sdLockingClause = clause }
```

Reply:
Added!